src/Dealer.ts

```diff
@@ -0,0 +1,300 @@
+      logger.debug({ usdLiabilityResult }, message)
+      return { ok: false, error: new Error(message) }
+    }
+
+    // Wallet usd balance is negative if actual liability,
+    // return additive inverse to deal with positive liability onward
+    const usdLiability = -usdLiabilityResult.value
+
+    const result = {} as UpdatedPositionAndLeverageResult
+
+    if (usdLiability < MINIMUM_POSITIVE_LIABILITY) {
+      logger.debug({ usdLiability }, "No liabilities to hedge, skipping the order loop")
+      result.updatePositionSkipped = true
+    } else {
+      logger.debug("starting with order loop")
+
+      const updatedPositionResult = await this.strategy.updatePosition(
+        usdLiability,
+        btcPriceInUsd,
+      )
+      result.updatedPositionResult = updatedPositionResult
+      if (updatedPositionResult.ok) {
+        const originalPosition = updatedPositionResult.value.originalPosition
+        const updatedPosition = updatedPositionResult.value.updatedPosition
+
+        logger.info(
+          { originalPosition, updatedPosition },
+          `The active ${activeStrategy} strategy was successful at UpdatePosition()`,
+        )
+        logger.debug(
+          { originalPosition },
+          `Position BEFORE ${activeStrategy} strategy executed UpdatePosition()`,
+        )
+        logger.debug(
+          { updatedPosition },
+          `Position AFTER ${activeStrategy} strategy executed UpdatePosition()`,
+        )
+      } else {
+        logger.error(
+          { updatedPosition: updatedPositionResult },
+          `The active ${activeStrategy} strategy failed during the UpdatePosition() execution`,
+        )
+      }
+    }
+
+    // Check for any in-flight fund transfer, and skip if not all completed
+    const dbCallResult = this.database.getPendingInFlightTransfers()
+    if (dbCallResult.ok && dbCallResult.value.size === 0) {
+      logger.debug("starting with rebalance loop")
+
+      const withdrawOnChainAddressResult =
+        await this.wallet.getWalletOnChainDepositAddress()
+      if (!withdrawOnChainAddressResult.ok || !withdrawOnChainAddressResult.value) {
+        const message = "WalletOnChainAddress is unavailable or invalid."
+        logger.debug({ withdrawOnChainAddressResult }, message)
+        return { ok: false, error: new Error(message) }
+      }
+      const withdrawOnChainAddress = withdrawOnChainAddressResult.value
+
+      const updatedLeverageResult = await this.strategy.updateLeverage(
+        usdLiability,
+        btcPriceInUsd,
+        withdrawOnChainAddress,
+        this.withdrawBookKeeping,
+        this.depositOnExchangeCallback,
+      )
+      result.updatedLeverageResult = updatedLeverageResult
+      if (updatedLeverageResult.ok) {
+        const updatedLeverage = updatedLeverageResult.value
+        logger.info(
+          { updatedLeverage },
+          `The active ${activeStrategy} strategy was successful at UpdateLeverage()`,
```

Comment:
`activeStrategy` should be passed as an argument in the first obj instead of the description.

Reply:
done, updated a bunch of them